src/context/index.ts

```diff
@@ -170,6 +171,28 @@ export class Context {
     return new Context({ polymeshApi, middlewareApi, keyring });
   }
 
+  /**
+   * Get if the current node is archive or not
+   */
+  public async isCurrentNodeArchive(): Promise<boolean> {
+    const {
+      polymeshApi: {
+        query: { balances, system },
+      },
+    } = this;
+
+    try {
+      const blockHash = await system.blockHash(0);
```

Comment:
use `numberToU32` here

Also, in `Proposal.minimumBondedAmount` we're using rpc to get the block hash from block number. We should use `system.blockHash` there too for consistency